From 0e106c095a4f0d1499be36fba94344890d10ed24 Mon Sep 17 00:00:00 2001
From: Fei Zhang <feizhang@codeaurora.org>
Date: Tue, 26 Nov 2013 10:10:20 +0800
Subject: [PATCH 063/176] QCamera2: fix Camcorder crash issue with Flash on

Camcorder will crash when enter with Flash on. Disable torch flash
support when in Camcorder mode.

Change-Id: I775daecb90bb69a7a8e6b7baf0b8b4da7a440b09
---
 QCamera2/HAL/QCameraParameters.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/QCamera2/HAL/QCameraParameters.cpp b/QCamera2/HAL/QCameraParameters.cpp
index 48924f1..263e06c 100644
--- a/QCamera2/HAL/QCameraParameters.cpp
+++ b/QCamera2/HAL/QCameraParameters.cpp
@@ -4730,7 +4730,7 @@ int32_t QCameraParameters::setFlash(const char *flashStr)
             ALOGV("%s: Setting Flash value %s", __func__, flashStr);
 
             if ( NULL != m_pTorch ) {
-                if ( value == CAM_FLASH_MODE_TORCH ) {
+                if ( value == CAM_FLASH_MODE_TORCH && !m_bRecordingHint_new) {
                     m_pTorch->prepareTorchCamera();
                 } else {
                     m_bReleaseTorchCamera = true;
@@ -7277,7 +7277,8 @@ int32_t QCameraParameters::commitParamChanges()
         m_bSceneTransitionAuto = false;
     }
 
-    if ( m_bReleaseTorchCamera && ( NULL != m_pTorch) ) {
+    if ( m_bReleaseTorchCamera && ( NULL != m_pTorch)
+        && !m_bRecordingHint_new ) {
         m_pTorch->releaseTorchCamera();
         m_bReleaseTorchCamera = false;
     }
-- 
1.8.3.1

